### ISSUE TITLE
1.19.2 Update

### ISSUE THREAD

**Issue by Ledix5** [Pokechu22/WorldDownloader#266]
Do you plan to update the mod to 1.19.2?

**Comment by Pokechu22** [comment]
I want to at some point, but do not have an ETA, and the update to 1.17+ is difficult due to internal changes to how the game works.

**Comment by StereoMadnessss** [comment]
also is there a fabric port

**Comment by milutinke** [comment]
If you have some time, you could make a simple manual about the code base and resources that need to be used, someone else might be able to contribute.
I also would like to, I already have some MC experience working with the MCC, but I have no clue about Forge, but given some time I probably would learn it.
What are the technical difficulties with the 1.17.1?

**Comment by The-Nyla** [comment]
I think they explained it in some other issues but if I Rember correctly it's mainly due to changes with forgegradle, the height limit changes in 1.17+, and also because of trying to build the mod to work across multiple versions at once

**Comment by Pokechu22** [comment]
Oh, sorry, yes, I missed this. The ForgeGradle part is the biggest one, but the others are also issues.

The mod's code itself is fairly well documented with javadoc comments; the highest-level overview is that mixins or patches are used to call code in WDLHooks.java, then WDLHooks.java does a bit of state management and calls WDLEvents.java, which calls out to a bunch of other code (including the code that handles copying inventories to block entities). Most state is stored in WDL.java (perhaps a bit too much state, is there). I don't think it should be too hard to navigate.

The build system is much less well documented, and I don't really remember how much of it works, unfortunately :/

**Comment by uyjulian** [comment]
Yeah mainly the problem is incompatible ForgeGradle versions and the maven repos and other remote dependencies may no longer be up.

I pretty much dropped off the Minecraft modding scene after LiteLoader stopped updating and I couldn’t be bothered to learn the new Fabric or whatever system is being used nowadays.

**Comment by The-Nyla** [comment]
There does exist a similar mod for fabric called “Chunkcopy” that’s open-source, it’s been updated to 1.19.2 but now seems to have been abandoned, if you know enough to continue support for that you could maybe ask the author for permission to do so. It has most of WDL’s features, but instead of saving the downloaded worlds to a singleplayer world, you have to make a world and run a command to paste the chunks. It likely wouldn’t be super hard to add an option to save the chunks to singleplayer worlds automatically like WDL.
